src/Stratis.Bitcoin.Features.Consensus/CoinViews/Coindb/LeveldbCoindb.cs

```diff
@@ -125,12 +126,44 @@ public void Initialize()
                 }
             }
 
+            EnsureCoinDatabaseIntegrity(chainTip);
+
             Block genesis = this.network.GetGenesis();
 
             if (this.GetTipHash() == null)
                 this.SetBlockHash(new HashHeightPair(genesis.GetHash(), 0));
 
-            this.logger.LogInformation("Coinview initialized with tip '{0}'.", this.persistedCoinviewTip);
+            this.logger.Info("Coinview initialized with tip '{0}'.", this.persistedCoinviewTip);
+        }
+
+        private void EnsureCoinDatabaseIntegrity(ChainedHeader chainTip)
+        {
+            this.logger.Info("Checking coin database integrity...");
+
+            var heightToCheck = chainTip.Height;
+
+            // Find the height up to where rewind data is stored above chain tip.
```

Comment:
There may be a more performant way to do this using the db enumerator methods.

Reply:
A enumerator wont really work here as I need to check for data at a specific height.